writer.go

```diff
@@ -19,7 +19,25 @@ type Writer struct {
 	framer    Framer
 	formatter Formatter
 
-	conn serverConn
+	conn          serverConn
+	needReconnect bool
+}
+
+// Poll the socket to find out if/when it closes
+func (w *Writer) poll() {
+	for {
+		if w.conn == nil {
+			w.needReconnect = true
+			return
+		}
+
+		b := make([]byte, 1)
+		_, err := w.conn.read(b)
```

Comment:
Got it, thank you. I'll push some updates as soon as I have a chance to work on this.